instrumentation/github.com/aws/aws-sdk-go-v2/otelaws/aws.go

```diff
@@ -0,0 +1,101 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otelaws
+
+import (
+	"context"
+
+	v2Middleware "github.com/aws/aws-sdk-go-v2/aws/middleware"
+	"github.com/aws/smithy-go/middleware"
+	smithyhttp "github.com/aws/smithy-go/transport/http"
+
+	"go.opentelemetry.io/contrib"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/semconv"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	tracerName = "go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-sdk-go-v2/otelaws"
+)
+
+type oTelMiddlewares struct {
+	tracer trace.Tracer
+}
+
+func (m oTelMiddlewares) initializeMiddleware(stack *middleware.Stack) error {
+	return stack.Initialize.Add(middleware.InitializeMiddlewareFunc("OtelSpanCreator", func(
+		ctx context.Context, in middleware.InitializeInput, next middleware.InitializeHandler) (
+		out middleware.InitializeOutput, metadata middleware.Metadata, err error) {
+
+		opts := []trace.SpanOption{
+			trace.WithSpanKind(trace.SpanKindClient),
+		}
+		ctx, span := m.tracer.Start(ctx, v2Middleware.GetServiceID(ctx), opts...)
```

Comment:
I have been implementing a similar tracing wrapper for the Datadog Go tracer directly, and I found that `GetServiceID` returned an empty string if `InitializeMiddlewareFunc` is used with `middleware.Before`. This was when appending the middleware to `aws.Config`.

Reply:
You are right. In my previous PR I added init middleware as `After` to get attributes from context. That is the reason I move `span.SetAttributes` to deserialize middleware.